app/src/test/java/com/duckduckgo/app/browser/refreshpixels/RefreshPixelSenderTest.kt

```diff
@@ -143,181 +115,28 @@ class RefreshPixelSenderTest {
-            appCoroutineScope = coroutineTestRule.testScope,
-            blockListPixelsPlugin = blockListPixelsPlugin,
-            dispatcherProvider = coroutineTestRule.testDispatcherProvider,
-        )
-
-        testee.sendMenuRefreshPixels()
-
-        verify(mockDao).updateRecentRefreshes(CURRENT_TIME - 20000, CURRENT_TIME)
-    }
-
-    @Test
-    fun whenSendPullToRefreshPixelsThenTimeBasedPixelsFired() = runTest {
-        val mockDao: RefreshDao = mock()
-
-        testee = DuckDuckGoRefreshPixelSender(
-            pixel = mockPixel,
-            dao = mockDao,
-            currentTimeProvider = mockCurrentTimeProvider,
-            appCoroutineScope = coroutineTestRule.testScope,
-            blockListPixelsPlugin = blockListPixelsPlugin,
-            dispatcherProvider = coroutineTestRule.testDispatcherProvider,
-        )
-
-        testee.sendPullToRefreshPixels()
-
-        verify(mockDao).updateRecentRefreshes(CURRENT_TIME - 20000, CURRENT_TIME)
-    }
-
-    @Test
-    fun whenSendCustomTabRefreshPixelThenTimeBasedPixelsFired() = runTest {
-        val mockDao: RefreshDao = mock()
-
-        testee = DuckDuckGoRefreshPixelSender(
-            pixel = mockPixel,
-            dao = mockDao,
-            currentTimeProvider = mockCurrentTimeProvider,
-            appCoroutineScope = coroutineTestRule.testScope,
-            blockListPixelsPlugin = blockListPixelsPlugin,
-            dispatcherProvider = coroutineTestRule.testDispatcherProvider,
-        )
-
-        testee.sendCustomTabRefreshPixel()
-
-        verify(mockDao).updateRecentRefreshes(CURRENT_TIME - 20000, CURRENT_TIME)
-    }
-
-    @Test
-    fun whenRefreshedOnceThenNoPixelFired() = runTest {
-        testee.sendMenuRefreshPixels()
-
-        verify(mockPixel, never()).fire(AppPixelName.RELOAD_TWICE_WITHIN_12_SECONDS)
-        verify(mockPixel, never()).fire(AppPixelName.RELOAD_THREE_TIMES_WITHIN_20_SECONDS)
-        assertTrue(refreshDao.all().size == 1)
-    }
-
-    @Test
-    fun whenRefreshedTwiceThenReloadTwicePixelFired() = runTest {
-        testee.sendMenuRefreshPixels()
-        testee.sendMenuRefreshPixels()
-
-        verify(mockPixel).fire(AppPixelName.RELOAD_TWICE_WITHIN_12_SECONDS)
-        verify(mockPixel, never()).fire(AppPixelName.RELOAD_THREE_TIMES_WITHIN_20_SECONDS)
-        assertNull(blockListPixelsPlugin.get2XRefresh())
-        assertTrue(refreshDao.all().size == 2)
-    }
-
-    @Test
-    fun whenRefreshedTwiceAndAssignedToExperimentThen2XRefreshPixelsFired() = runTest {
+    fun whenRefreshedTwiceOrThriceAndAssignedToExperimentThen2XRefreshPixelsFired() = kotlinx.coroutines.test.runTest {
         assignToExperiment()
-        testee.sendMenuRefreshPixels()
-        testee.sendMenuRefreshPixels()
+        testee.sendBreakageRefreshPixels(setOf(2, 3))
 
-        blockListPixelsPlugin.get2XRefresh()!!.getPixelDefinitions().forEach {
+        blockListPixelsPlugin.get2XRefresh()?.getPixelDefinitions()?.forEach {
```

Comment:
Go back to `!!` I want this test to fail if `get2Refresh` returns null and this would bypass it.